src/c_datove_typy.md

```diff
@@ -1,6 +1,127 @@
+reprezentovat 256 různých hodnot. Pokud budeme interpretovat toto číslo se znaménkem, tak může uchovávat
+hodnoty -128 až 127. Pokud ho budeme interpretovat bez znaménka, tak může uchovávat hodnoty 0 až 255.
+
+C obsahuje několik základních typů celočíselných proměnných, které se liší v tom, kolik mají bytů a
+jestli jsou znaménkové nebo ne. Pokud před název typu napíšeme `signed`, bude se jednat o znaménkový
+typ, pokud použijeme `unsigned`, tak použijeme typ bez znaménka. Většina typů je implicitně se
+znaménkem, tj. `int` je to samé jako `signed int`. V následující tabulce je seznam nejčastějších
+celočíselných typů[^2]:
+
+| Název | Počet bytů | Rozsah hodnot | Se znaménkem |
+|---|:---:|:---:|:---:|
+| `char` nebo<br />`signed char` | 1 | [-128; 127] | <i class="fa fa-check"></i> |
+| `unsigned char` | 1 | [0; 255] | <i class="fa fa-times"></i> |
+| `short` nebo<br />`signed short` | 2 | [-32 768; 32 767] | <i class="fa fa-check"></i> |
+| `unsigned short` | 2 | [0; 65 535] | <i class="fa fa-times"></i> |
+| **`int`** nebo<br />`signed int` | 4 | [-2 147 483 648; 2 147 483 647] | <i class="fa fa-check"></i> |
+| `unsigned int` | 4 | [0; 4 294 967 295] | <i class="fa fa-times"></i> |
+| `long` nebo<br />`signed long` | 8 | [-9 223 372 036 854 775 808;<br />9 223 372 036 854 775 807] | <i class="fa fa-check"></i> |
+| `unsigned long` | 8 | [0; 18 446 744 073 709 551 615] | <i class="fa fa-times"></i> |
+
+[^2]: Počet bytů (a znaménkovost u typu `char`) záleží na použitém
+operačním systému a překladači. Zde jsou uvedeny hodnoty, se kterými se můžete
+nejčastěji setkat na 64-bitovém Linuxovém systému s překladačem GCC při použití
+[dvojkového doplňku](https://cs.wikipedia.org/wiki/Dvojkov%C3%BD_dopln%C4%9Bk).
+
+Každý vestavěný datový typ (`char`, `short`, `int`) a modifikátor znaménkovosti (`signed`, `unsigned`)
+je zároveň klíčovým slovem.
+
+Pokud ze začátku nebudete vědět, který typ zvolit, tak pro základní aritmetické operace používejte
+ze začátku typy se znaménkem s 4 byty, tedy `int`. Tento typ je také implicitně použit když v programu
+použijete číselný výraz, například výraz `1` má datový typ `int`[^3].
+
+[^3]: Pouze pokud by výraz nešel reprezentovat typem `int`, použije se číselný typ s více byty.
+
+> Typ `char` je speciální v tom, že zároveň běžně reprezentuje textové znaky v
+> [ASCII](https://www.asciitable.com/) kódování. Více o reprezentaci textu v programech se dozvíte
+> v sekci o [řetězcích](c_retezce.md).
+
+#### Operace s číselnými typy
+C umožňuje provádět operace nad vestavěnými datovými typy pomocí tzv. **operátorů**. Při práci s
+výrazy celočíselných typů lze provádět běžné aritmetické operace `+`, `-`, `/`, `*` nebo `%` (zbytek
+po dělení). Například `5 + 8` nebo `2 * 16` tak bude obvykle fungovat tak, jak byste očekávali. Je si
+ale třeba dát pozor na několik zrádných věcí:
+
+- Při dělení dvou celočíselných čísel pomocí operátoru `/` dochází k celočíselnému dělení, tj. například
```

Comment:
tu by to mozna chtelo interaktivni ukazku, at se to lepe zapamatuje... tohle je nejvetsi problem:)

Reply:
good point, přidáno